### DISCUSSION TITLE
xmlsec1- SignatureValue splitting across multiple lines - change setting to output as single line

### DISCUSSION THREAD

**Discussion by mueller68** [lsh123/xmlsec#705] [Q&A] [answered]
I am trying to sign an XML document for use in a program which is closed source and I cannot change how it is written.

Existing xml-signed files have all of the values for the <SignatureValue> on a single line.

When I generate a signed document using xmlsec1 on Fedora linux, it is generating the file, but the <SignatureValue> is split across 3 lines with the line feed at the end.  I believe that manually changing it to a single line will invalidate it when it does the check, so I’m trying to generate it from xmlsec1 without the line feed.  So I am trying to see if there is a setting or a way I can get it to output into a single line.

I am using RSA1024 Key Length

For example, it is outputting with these line feeds:
 
![image](https://github.com/lsh123/xmlsec/assets/140929560/a247f695-6061-4901-8cdf-1c9e64b687bb)

What I want is the output to look like:
 
![image](https://github.com/lsh123/xmlsec/assets/140929560/655cad4e-fceb-4809-9821-95d74f762da1)

What can I do to prevent it from splitting across multiple lines because of the length of the string?

**Comment by lsh123** [comment, marked as answer]
I've added a command line option `--base64-line-size` in PR https://github.com/lsh123/xmlsec/pull/706 and you can set it to some very large value (e.g. `--base64-line-size 123456789`) to disable line breaks.

It will be included in the next release or you can build the xmlsec library and command line tool from master after I merge the PR shortly.
